src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.ConvertToInteger.cs

```diff
@@ -32,23 +32,23 @@ public static void ConvertToInteger<TFrom, TTo>(ReadOnlySpan<TFrom> source, Span
             public static TTo Invoke(TFrom x) => TFrom.ConvertToInteger<TTo>(x);
 
             public static bool Vectorizable =>
-                (typeof(TFrom) == typeof(float) && typeof(TTo) == typeof(int)) ||
-                (typeof(TFrom) == typeof(float) && typeof(TTo) == typeof(uint)) ||
-                (typeof(TFrom) == typeof(double) && typeof(TTo) == typeof(long)) ||
-                (typeof(TFrom) == typeof(double) && typeof(TTo) == typeof(ulong));
+                (typeof(TFrom) == typeof(float) && IsInt32Like<TTo>()) ||
+                (typeof(TFrom) == typeof(float) && IsUInt32Like<TTo>()) ||
+                (typeof(TFrom) == typeof(double) && IsInt64Like<TTo>()) ||
+                (typeof(TFrom) == typeof(double) && IsUInt64Like<TTo>());
```

Comment:
This could be simplified to `Vector128<TFrom>.IsSupported && Vector128<TTo>.IsSupported && (sizeof(TFrom) == sizeof(TTo))`, which would allow the JIT to continue seeing it as a "constant" and not be dependent on the inliner